<launch>

    <!-- Parameters
    <arg name="model" default="$(find div_datalogger_postprocess)/urdf/DataLogger.urdf" />
      <arg name="gui" default="true" />-->
    <arg name="rvizconfig" default="$(find div_datalogger_postprocess)/rviz/urdf.rviz" />
    <param name="use_sim_time" value="true"/>
    <node name="rosbag_play" pkg="rosbag"  type="play" args="/home/christian-nils/CN/Program/python/test2/sync_filtered_test.bag -l --clock"/>

    <!-- TF broadcasters-->
    <node name="imu_to_tf" pkg="div_datalogger_postprocess" type="bike_imu_to_tf.py" respawn="true" output="screen"/>
    <node pkg="tf" type="static_transform_publisher" name="odom_2_footprint" args="0 0 0 0 0 0 odom base_footprint 100"/>
    <node pkg="tf" type="static_transform_publisher" name="imu0_2_handlebar" args="0 1.10 -1.00 0 0 0 imu0_link handlebar_link 100"/>
    <node pkg="tf" type="static_transform_publisher" name="imu0_2_laser" args=".135 -.25 -.44 -.44 .1 3.16 imu0_link laser 100"/>


    <!-- should be replaced by EKF
    <node pkg="tf" type="static_transform_publisher" name="odom_2_base_footprint" args="0 0 0 0 0 0 odom base_footprint 100"/>

      <node pkg="tf" type="static_transform_publisher" name="base_footprint_2_base_stabilized" args="0 0 0.86 0 0 0 base_footprint base_stabilized 100"/>
      <node pkg="tf" type="static_transform_publisher" name="base_link_2_gps" args="0 0 -0.11 0 0 3.14 base_footprint gps 100"/>
      <node pkg="tf" type="static_transform_publisher" name="base_link_2_laser" args="0 0 -0.11 3.14 0 3.14 base_link laser 100"/>-->

    <!-- EKF -->
    <!--<node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true" output="screen">
        <rosparam command="load" file="$(find robot_localization)/params/ekf_template.yaml" />
      </node>-->

    <!-- GPS -->
    <!--
      <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" clear_params="true" output="screen">
        <rosparam command="load" file="$(find robot_localization)/params/navsat_transform_template.yaml" />
          <remap from="imu/data" to="imu/data"/>
          <remap from="odometry/filtered" to="odometry/filtered"/>
          <remap from="gps/fix" to="fix"/>
      </node>
     -->

    <!--<node pkg="laser_scan_matcher" type="laser_scan_matcher_node" name="laser_scan_matcher_node" output="screen">
        <remap from="/imu/data" to="/imu0/data"/>
        <param name="use_odom" value="false"/>
        <param name="fixed_frame" value="odom"/>
        <param name="base_frame" value="base_footprint"/>
        <param name="max_iterations" value="20"/>
  </node>-->
    <!-- for package laseÅ• assembler-->
    <node type="laser_scan_assembler" pkg="laser_assembler" name="my_assembler" output="screen">
      <remap from="scan" to="/scan_corrected"/>
      <param name="max_scans" type="int" value="1000" />
      <param name="fixed_frame" type="string" value="odom" />
      <param name="ignore_laser_skew" type="bool" value="true" />
    </node>
    <node type="periodic_snapshotter" name="periodic_snapshotter" pkg="laser_assembler" output="screen"></node>
-->
    <!-- RVIZ-->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />

</launch>
